core/headerchain.go

```diff
@@ -126,30 +131,33 @@ func (hc *HeaderChain) WriteHeader(header *types.Header) (status WriteStatus, er
 			DeleteCanonicalHash(hc.chainDb, i)
 		}
 		// Overwrite any stale canonical number assignments
-		head := hc.GetHeader(header.ParentHash)
-		for GetCanonicalHash(hc.chainDb, head.Number.Uint64()) != head.Hash() {
-			WriteCanonicalHash(hc.chainDb, head.Hash(), head.Number.Uint64())
-			head = hc.GetHeader(head.ParentHash)
+		head, headHash := hc.GetHeader(header.ParentHash), header.ParentHash
```

Comment:
Could we please stop using these assignments? (I honestly don't understand why Go permitted this).

```go
var (
    head = ...
    headHash = ...
)
```

makes code so much cleaner.